measurement/dash_apps/data_report.py

```diff
@@ -111,26 +117,42 @@ def update_graph(
     variable: str,
     start_time: str,
     end_time: str,
-) -> px.line:
+    relayout_data: dict,
+    callback_context,
+) -> go.Figure:
+    ctx = callback_context
+    triggered_id = ctx.triggered[0]["prop_id"].split(".")[0] if ctx.triggered else ""
+
+    # This is cached, so it's not a big deal to call it multiple times
+    data = get_report_data_from_db(
+        station=station,
+        variable=variable,
+        start_time=start_time,
+        end_time=end_time,
+        report_type=temporality,
+    )
+    if data.empty:
+        return create_empty_plot()
+
+    if triggered_id == "data_report_graph" and "xaxis.range[0]" in relayout_data:
+        start = relayout_data["xaxis.range[0]"]
+        end = relayout_data["xaxis.range[1]"]
+        data = data[(data["time"] >= start) & (data["time"] <= end)]
+
     try:
-        data = get_report_data_from_db(
-            station=station,
-            variable=variable,
-            start_time=start_time,
-            end_time=end_time,
-            report_type=temporality,
-        )
+        every = max(1, len(data) // MAX_POINTS)
+        resampled = data.iloc[::every]
+
```

Comment:
When I was attempting this, I put all of this in the plots.py file, I guess that's why it was not working